File: 02/solution.js
const fs = require('fs')

const [xf, yf, aimf] = fs.readFileSync('input', 'utf8').split("\n") // Load input
    .filter(l => l) // Strip empty lines
    .map(c => c.split(' ')) // Split command into direction and value
    .map(([direction, valueString]) => [direction, parseInt(valueString, 10)]) // Parse value into int
    .reduce(([x, y, aim], [direction, value]) => ({
            'forward': [(x + value), (y + aim * value), aim],
            'up': [x, y, (aim - value)],
            'down': [x, y, (aim + value)],
        }[direction]), [0, 0, 0])

console.log(xf * yf)

File: 02/solution.clj
(require '[clojure.string :as str])

(let [[xf yf aimf]
  (->> (str/split-lines (slurp "input")) ;; Load input
    (filter not-empty) ;; Strip empty lines
    (map #(str/split % #" ")) ;; Split command into direction and value
    (map #(vector (first %) (Integer/parseInt (second %)))) ;; Parse value into int
    (reduce (fn [[x y aim] [direction value]]
      (case direction
        "forward" [(+ x value) (+ y (* aim value)) aim]
        "up" [x y (- aim value)]
        "down" [x y (+ aim value)])) [0 0 0]))]
  (print (* xf yf)))
